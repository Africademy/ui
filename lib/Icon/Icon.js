"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var styles = _interopRequireWildcard(require("./style"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  ", ";\n  ", ";\n\n  vertical-align: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var StyledIcon = _styledComponents.default.svg(_templateObject(), styles.base, function (props) {
  return styles[props.size];
}, function (props) {
  return props.verticalAlign ? props.verticalAlign : null;
});

var Icon = function Icon(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return _react.default.createElement(StyledIcon, props, children);
};

Icon.propTypes = {
  /** The size of the icon. Can be 'small', 'medium', or 'large' */
  size: _propTypes.default.oneOf(['smallest', 'small', 'medium', 'large']),

  /** The `vertical-align` CSS value */
  verticalAlign: _propTypes.default.string,

  /** */
  children: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.node]).isRequired
};
Icon.defaultProps = {
  size: 'medium',
  verticalAlign: ''
};
var _default = Icon;
exports.default = _default;