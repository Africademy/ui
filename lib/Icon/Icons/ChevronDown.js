"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _createIconComponent = _interopRequireDefault(require("../utils/createIconComponent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** WARNING
 * This file is automatically generated by the `npm run gen:icons` command.
 * ALL changes will be overwritten.
 */
var ChevronDownIcon = (0, _createIconComponent.default)({
  content: _react.default.createElement("g", null, _react.default.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11.146 6.854A.5.5 0 0 0 10.793 6H5.207a.5.5 0 0 0-.353.854l2.792 2.792a.5.5 0 0 0 .708 0l2.792-2.792z"
  }))
});
ChevronDownIcon.displayName = 'ChevronDownIcon';
var _default = ChevronDownIcon;
exports.default = _default;