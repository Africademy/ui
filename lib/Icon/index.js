"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ArrowDown", {
  enumerable: true,
  get: function get() {
    return _ArrowDown2.default;
  }
});
Object.defineProperty(exports, "ArrowLeft", {
  enumerable: true,
  get: function get() {
    return _ArrowLeft2.default;
  }
});
Object.defineProperty(exports, "ArrowRight", {
  enumerable: true,
  get: function get() {
    return _ArrowRight2.default;
  }
});
Object.defineProperty(exports, "ArrowUp", {
  enumerable: true,
  get: function get() {
    return _ArrowUp2.default;
  }
});
Object.defineProperty(exports, "Attach", {
  enumerable: true,
  get: function get() {
    return _Attach2.default;
  }
});
Object.defineProperty(exports, "Camera", {
  enumerable: true,
  get: function get() {
    return _Camera2.default;
  }
});
Object.defineProperty(exports, "ChartLine", {
  enumerable: true,
  get: function get() {
    return _ChartLine2.default;
  }
});
Object.defineProperty(exports, "Checkmark", {
  enumerable: true,
  get: function get() {
    return _Checkmark2.default;
  }
});
Object.defineProperty(exports, "ChevronDown", {
  enumerable: true,
  get: function get() {
    return _ChevronDown2.default;
  }
});
Object.defineProperty(exports, "ChevronUp", {
  enumerable: true,
  get: function get() {
    return _ChevronUp2.default;
  }
});
Object.defineProperty(exports, "Clock", {
  enumerable: true,
  get: function get() {
    return _Clock2.default;
  }
});
Object.defineProperty(exports, "Compare", {
  enumerable: true,
  get: function get() {
    return _Compare2.default;
  }
});
Object.defineProperty(exports, "Copy", {
  enumerable: true,
  get: function get() {
    return _Copy2.default;
  }
});
Object.defineProperty(exports, "Emoji", {
  enumerable: true,
  get: function get() {
    return _Emoji2.default;
  }
});
Object.defineProperty(exports, "Facebook", {
  enumerable: true,
  get: function get() {
    return _Facebook2.default;
  }
});
Object.defineProperty(exports, "Flag", {
  enumerable: true,
  get: function get() {
    return _Flag2.default;
  }
});
Object.defineProperty(exports, "Folder", {
  enumerable: true,
  get: function get() {
    return _Folder2.default;
  }
});
Object.defineProperty(exports, "Gear", {
  enumerable: true,
  get: function get() {
    return _Gear2.default;
  }
});
Object.defineProperty(exports, "GooglePlus", {
  enumerable: true,
  get: function get() {
    return _GooglePlus2.default;
  }
});
Object.defineProperty(exports, "Instagram", {
  enumerable: true,
  get: function get() {
    return _Instagram2.default;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link2.default;
  }
});
Object.defineProperty(exports, "LinkedIn", {
  enumerable: true,
  get: function get() {
    return _LinkedIn2.default;
  }
});
Object.defineProperty(exports, "List", {
  enumerable: true,
  get: function get() {
    return _List2.default;
  }
});
Object.defineProperty(exports, "Load", {
  enumerable: true,
  get: function get() {
    return _Load2.default;
  }
});
Object.defineProperty(exports, "Location", {
  enumerable: true,
  get: function get() {
    return _Location2.default;
  }
});
Object.defineProperty(exports, "Locked", {
  enumerable: true,
  get: function get() {
    return _Locked2.default;
  }
});
Object.defineProperty(exports, "Message", {
  enumerable: true,
  get: function get() {
    return _Message2.default;
  }
});
Object.defineProperty(exports, "Messenger", {
  enumerable: true,
  get: function get() {
    return _Messenger2.default;
  }
});
Object.defineProperty(exports, "More", {
  enumerable: true,
  get: function get() {
    return _More2.default;
  }
});
Object.defineProperty(exports, "Pause", {
  enumerable: true,
  get: function get() {
    return _Pause2.default;
  }
});
Object.defineProperty(exports, "Pencil", {
  enumerable: true,
  get: function get() {
    return _Pencil2.default;
  }
});
Object.defineProperty(exports, "People", {
  enumerable: true,
  get: function get() {
    return _People2.default;
  }
});
Object.defineProperty(exports, "Person", {
  enumerable: true,
  get: function get() {
    return _Person2.default;
  }
});
Object.defineProperty(exports, "Pinterest", {
  enumerable: true,
  get: function get() {
    return _Pinterest2.default;
  }
});
Object.defineProperty(exports, "Placeholder", {
  enumerable: true,
  get: function get() {
    return _Placeholder2.default;
  }
});
Object.defineProperty(exports, "Plus", {
  enumerable: true,
  get: function get() {
    return _Plus2.default;
  }
});
Object.defineProperty(exports, "Refresh", {
  enumerable: true,
  get: function get() {
    return _Refresh2.default;
  }
});
Object.defineProperty(exports, "Return", {
  enumerable: true,
  get: function get() {
    return _Return2.default;
  }
});
Object.defineProperty(exports, "Search", {
  enumerable: true,
  get: function get() {
    return _Search2.default;
  }
});
Object.defineProperty(exports, "Sidebar", {
  enumerable: true,
  get: function get() {
    return _Sidebar2.default;
  }
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function get() {
    return _Tag2.default;
  }
});
Object.defineProperty(exports, "Twitter", {
  enumerable: true,
  get: function get() {
    return _Twitter2.default;
  }
});
Object.defineProperty(exports, "Unlocked", {
  enumerable: true,
  get: function get() {
    return _Unlocked2.default;
  }
});
Object.defineProperty(exports, "Warning", {
  enumerable: true,
  get: function get() {
    return _Warning2.default;
  }
});

var _ArrowDown2 = _interopRequireDefault(require("./Icons/ArrowDown"));

var _ArrowLeft2 = _interopRequireDefault(require("./Icons/ArrowLeft"));

var _ArrowRight2 = _interopRequireDefault(require("./Icons/ArrowRight"));

var _ArrowUp2 = _interopRequireDefault(require("./Icons/ArrowUp"));

var _Attach2 = _interopRequireDefault(require("./Icons/Attach"));

var _Camera2 = _interopRequireDefault(require("./Icons/Camera"));

var _ChartLine2 = _interopRequireDefault(require("./Icons/ChartLine"));

var _Checkmark2 = _interopRequireDefault(require("./Icons/Checkmark"));

var _ChevronDown2 = _interopRequireDefault(require("./Icons/ChevronDown"));

var _ChevronUp2 = _interopRequireDefault(require("./Icons/ChevronUp"));

var _Clock2 = _interopRequireDefault(require("./Icons/Clock"));

var _Compare2 = _interopRequireDefault(require("./Icons/Compare"));

var _Copy2 = _interopRequireDefault(require("./Icons/Copy"));

var _Emoji2 = _interopRequireDefault(require("./Icons/Emoji"));

var _Facebook2 = _interopRequireDefault(require("./Icons/Facebook"));

var _Flag2 = _interopRequireDefault(require("./Icons/Flag"));

var _Folder2 = _interopRequireDefault(require("./Icons/Folder"));

var _Gear2 = _interopRequireDefault(require("./Icons/Gear"));

var _GooglePlus2 = _interopRequireDefault(require("./Icons/GooglePlus"));

var _Instagram2 = _interopRequireDefault(require("./Icons/Instagram"));

var _Link2 = _interopRequireDefault(require("./Icons/Link"));

var _LinkedIn2 = _interopRequireDefault(require("./Icons/LinkedIn"));

var _List2 = _interopRequireDefault(require("./Icons/List"));

var _Load2 = _interopRequireDefault(require("./Icons/Load"));

var _Location2 = _interopRequireDefault(require("./Icons/Location"));

var _Locked2 = _interopRequireDefault(require("./Icons/Locked"));

var _Message2 = _interopRequireDefault(require("./Icons/Message"));

var _Messenger2 = _interopRequireDefault(require("./Icons/Messenger"));

var _More2 = _interopRequireDefault(require("./Icons/More"));

var _Pause2 = _interopRequireDefault(require("./Icons/Pause"));

var _Pencil2 = _interopRequireDefault(require("./Icons/Pencil"));

var _People2 = _interopRequireDefault(require("./Icons/People"));

var _Person2 = _interopRequireDefault(require("./Icons/Person"));

var _Pinterest2 = _interopRequireDefault(require("./Icons/Pinterest"));

var _Placeholder2 = _interopRequireDefault(require("./Icons/Placeholder"));

var _Plus2 = _interopRequireDefault(require("./Icons/Plus"));

var _Refresh2 = _interopRequireDefault(require("./Icons/Refresh"));

var _Return2 = _interopRequireDefault(require("./Icons/Return"));

var _Search2 = _interopRequireDefault(require("./Icons/Search"));

var _Sidebar2 = _interopRequireDefault(require("./Icons/Sidebar"));

var _Tag2 = _interopRequireDefault(require("./Icons/Tag"));

var _Twitter2 = _interopRequireDefault(require("./Icons/Twitter"));

var _Unlocked2 = _interopRequireDefault(require("./Icons/Unlocked"));

var _Warning2 = _interopRequireDefault(require("./Icons/Warning"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }