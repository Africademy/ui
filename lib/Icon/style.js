"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.large = exports.medium = exports.small = exports.smallest = exports.base = void 0;

var _styledComponents = require("styled-components");

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  width: 22px;\n  height: 22px;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  width: 16px;\n  height: 16px;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  width: 12px;\n  height: 12px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  width: 8px;\n  height: 8px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n  fill: currentcolor;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var base = (0, _styledComponents.css)(_templateObject());
exports.base = base;
var smallest = (0, _styledComponents.css)(_templateObject2());
exports.smallest = smallest;
var small = (0, _styledComponents.css)(_templateObject3());
exports.small = small;
var medium = (0, _styledComponents.css)(_templateObject4());
exports.medium = medium;
var large = (0, _styledComponents.css)(_templateObject5());
exports.large = large;