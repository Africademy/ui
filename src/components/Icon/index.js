export Analyze from './Icons/Analyze';
export ArrowDown from './Icons/ArrowDown';
export ArrowLeft from './Icons/ArrowLeft';
export ArrowRight from './Icons/ArrowRight';
export Attach from './Icons/Attach';
export Bell from './Icons/Bell';
export Buffer from './Icons/Buffer';
export Checkmark from './Icons/Checkmark';
export ChevronDown from './Icons/ChevronDown';
export ChevronUp from './Icons/ChevronUp';
export Clock from './Icons/Clock';
export Compare from './Icons/Compare';
export Copy from './Icons/Copy';
export Cross from './Icons/Cross';
export Direct from './Icons/Direct';
export Dollar from './Icons/Dollar';
export Email from './Icons/Email';
export Emoji from './Icons/Emoji';
export Facebook from './Icons/Facebook';
export Flag from './Icons/Flag';
export Flash from './Icons/Flash';
export Folder from './Icons/Folder';
export Folders from './Icons/Folders';
export Gear from './Icons/Gear';
export Google from './Icons/Google';
export GooglePlus from './Icons/GooglePlus';
export Hashtag from './Icons/Hashtag';
export Hide from './Icons/Hide';
export Inbox from './Icons/Inbox';
export Info from './Icons/Info';
export Instagram from './Icons/Instagram';
export Link from './Icons/Link';
export LinkedIn from './Icons/LinkedIn';
export List from './Icons/List';
export Load from './Icons/Load';
export Location from './Icons/Location';
export Locked from './Icons/Locked';
export Messenger from './Icons/Messenger';
export OpenNew from './Icons/OpenNew';
export Pause from './Icons/Pause';
export Person from './Icons/Person';
export Pinterest from './Icons/Pinterest';
export Placeholder from './Icons/Placeholder';
export Publish from './Icons/Publish';
export QrCode from './Icons/QrCode';
export Refresh from './Icons/Refresh';
export Reply from './Icons/Reply';
export Return from './Icons/Return';
export Search from './Icons/Search';
export Shopify from './Icons/Shopify';
export Swap from './Icons/Swap';
export Tag from './Icons/Tag';
export Trash from './Icons/Trash';
export Twitter from './Icons/Twitter';
export Unknown from './Icons/Unknown';
export Unlocked from './Icons/Unlocked';
export View from './Icons/View';
export Warning from './Icons/Warning';
export Website from './Icons/Website';
export Whatsapp from './Icons/Whatsapp';
export Youtube from './Icons/Youtube';
