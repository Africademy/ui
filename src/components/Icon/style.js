import { css } from 'styled-components';

export const base = css`
  fill: currentColor;
`;


export const chevron = css`
  margin-right: -5px;
`;

export const small = css`
  width: 12px;
  height: 12px;
`;

export const medium = css`
  width: 16px;
  height: 16px;
`;

export const large = css`
  width: 22px;
  height: 22px;
`;
