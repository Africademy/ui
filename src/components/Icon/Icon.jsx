import React from 'react';
import PropTypes from 'prop-types';
import style from 'styled-components';
import * as Styles from './style';

const IconStyled = style.svg`  
  ${Styles.base};
  ${props => Styles[props.size]};
  ${props => Styles[props.disabled ? 'disabled' : props.type]};
`;

const GroupStyled = style.g`  
  ${props => Styles[`${props.type}Group`]};
`;

const Icon = ({ children, size, type }) => (
  <IconStyled
    size={size}
    type={type}
    viewBox="0 0 16 16"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <GroupStyled type={type}>
      {children}
    </GroupStyled>
  </IconStyled>
);

Icon.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.oneOf('small', 'medium', 'large'),
  type: PropTypes.oneOf('primary', 'secondary'),
};

Icon.defaultProps = {
  size: 'medium',
  type: 'secondary',
};

export default Icon;
