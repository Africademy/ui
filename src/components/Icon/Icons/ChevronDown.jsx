import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';
import { StyledIcon } from './style';

const ChevronDown = ({ type, size }) => (
  <Icon type={type} size={size}>
    <StyledIcon type={type} fillRule="evenodd" clipRule="evenodd" d="M7.14645 0.853553C7.46143 0.53857 7.23835 0 6.79289 0L1.20711 0C0.761654 0 0.538571 0.538571 0.853553 0.853553L3.64645 3.64645C3.84171 3.84171 4.15829 3.84171 4.35355 3.64645L7.14645 0.853553Z" fill="#0F0B0A" />
  </Icon>
);

ChevronDown.propTypes = {
  size: PropTypes.oneOf('small', 'medium', 'large'),
  type: PropTypes.oneOf('primary', 'secondary'),
};

ChevronDown.defaultProps = {
  size: 'medium',
  type: 'secondary',
};


export default ChevronDown;
