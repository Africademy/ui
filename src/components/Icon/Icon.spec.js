/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import snap from 'jest-auto-snapshots';
import renderer from 'react-test-renderer';
import Icon from './Icon';
import ChevronDown from './Icons/ChevronDown';
import 'jest-styled-components';

snap(Icon, './Icon.jsx');

test('Button renders styles properly', () => {
  const tree = renderer.create(<Icon type="primary" size="medium"><ChevronDown /></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('width', '18px');
});
