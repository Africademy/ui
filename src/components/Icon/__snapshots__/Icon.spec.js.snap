// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Icon Matches snapshot when passed all props 1`] = `
<ForwardRef
  size="small"
  type="primary"
  version="1.1"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <ForwardRef
    type="primary"
  >
    <NodeFixture />
  </ForwardRef>
</ForwardRef>
`;

exports[`jest-auto-snapshots > Icon Matches snapshot when passed only required props 1`] = `
<ForwardRef
  size="medium"
  type="secondary"
  version="1.1"
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <ForwardRef
    type="secondary"
  >
    <NodeFixture />
  </ForwardRef>
</ForwardRef>
`;
